### ISSUE TITLE
addSrcFromDir needs _safe_path() on Windows

### ISSUE THREAD

**Issue by oroppas** [graalvm/mx#158]
```
Archiving GRAAL_COMPILER_WHITEBOX_MICRO_BENCHMARKS... [dependency org.graalvm.compiler.virtual.bench updated]
Traceback (most recent call last):
  File "C:/msys64/home/cd4mcq/packages/graal/mx/mx.py", line 18330, in <module>
    main()
  File "C:/msys64/home/cd4mcq/packages/graal/mx/mx.py", line 18311, in main
    retcode = c(command_args)
  File "C:/msys64/home/cd4mcq/packages/graal/mx/mx.py", line 12228, in build
    t.execute()
  File "C:/msys64/home/cd4mcq/packages/graal/mx/mx.py", line 810, in execute
    self.build()
  File "C:/msys64/home/cd4mcq/packages/graal/mx/mx.py", line 1722, in build
    self.subject.make_archive()
  File "C:/msys64/home/cd4mcq/packages/graal/mx/mx.py", line 1486, in make_archive
    addSrcFromDir(srcDir, archivePrefix, arcnameCheck=overlay_check)
  File "C:/msys64/home/cd4mcq/packages/graal/mx/mx.py", line 1376, in addSrcFromDir
    info = zipfile.ZipInfo(arcname, time.localtime(os.path.getmtime(join(root, f)))[:6])
  File "C:\msys64\home\cd4mcq\pypy2-latest-win32\lib-python\2.7\genericpath.py", line 62, in getmtime
    return os.stat(filename).st_mtime
WindowsError: [Error 3] The system cannot find the path specified: C:\msys64\home\cd4mcq\packages\graal\graal\compiler\mxbuild\src\org.graalvm.compiler.microbenchmarks\src_gen\org\graalvm\compiler\microbenchmarks\graal\generated\SchedulePhaseBenchmark_ScheduleEarliestIterative_EARLIEST_WITH_GUARD_ORDER_OPTIMAL_jmhType_B1.java
```

**Comment by dougxc** [comment]
Hi @oroppas  thanks for opening this issue. Maybe you would be able to submit a PR to fix this? I don't have a Windows machine handy to test this.

**Comment by dougxc** [comment]
@pejovica should this now be fixed by other Windows related changes to mx?

**Comment by pejovica** [comment]
Yes, it should be fixed by #171.

**Comment by dougxc** [comment]
Thanks. @oroppas please re-open if you still see this problem.

**Comment by oroppas** [comment]
Thank you @pejovica, @dougxc.
Yes, it's fixed by #171.
